### ISSUE TITLE
Suggestion: dplyr::summarize, dplyr::mutate, dplyr::filter - treat new-line as comma

### ISSUE THREAD

**Issue by EconomiCurtis** [hadley/dplyr#1279]
tl:dr, curious what you think about having certain dplyr functions recognize new-lines as commas?

----
When I use functions like summarize, mutate, filter, arrange etc, I find myself separating arguments by new lines.  For example with mutate and multiple name-value pairs of expressions from the  [dplyr::vignette](https://cran.rstudio.com/web/packages/dplyr/vignettes/introduction.html#add-new-columns-with-mutate),

     mutate(flights, gain = arr_delay - dep_delay, speed = distance / air_time * 60)
is not as nice as

     mutate(flights,
        gain = arr_delay - dep_delay,
        speed = distance / air_time * 60
     )

Now, as I add many new lines it is quite easy to forget the comma separating expressions, (leading to the " unexpected symbol in" error, and occasionally the "not a promise" error. 

I was thinking things could look a lot more elegant if in this situation, dplyr recognized a new-line as a comma. 

     mutate(flights,
        gain = arr_delay - dep_delay  #note no comma
        speed = distance / air_time * 60
     )

Small idea I know. But I make this mistake daily. Sorry to bother. 

Motivation: 
- I think this is a good case in terms of general ease of use, python-type simplification, smooth handling of typical human errors.
- People could continue to use commas. That is, I am pretty sure this change would be backwards compatible, it wouldn't break anyone's older code that relies on dplyr. 
- If one wanted an expression to go over more than one line, they could use parentheses.

**Comment by hadley** [comment]
It would be quite hard to implement this, and it add a new type of syntax, so while I agree the result would be nice, I don't think it's worth the cost.
